Code/RDGeneral/BoostStartInclude.h

```diff
@@ -44,7 +44,11 @@
 #pragma GCC diagnostic ignored "-Wtype-limits"
 #pragma GCC diagnostic ignored "-Wreorder"
 #pragma GCC diagnostic ignored "-Wunused"
-#pragma GCC diagnostic ignored "-Wunused-local-typedef"
+#if defined(__apple_build_version__)
```

Comment:
This might have been added prior to the -Wunused-local-typedef in CMakeLIsts.txt

Note that I don't really think that it should be in CMakeLists.txt as this hides our issues that should probably be fixed.